usr/share/rear/wrapup/default/970_ssh_agent_stop.sh

```diff
@@ -0,0 +1,9 @@
+# ssh-agent shutdown during 'rear recover'
+#
+# This file is part of Relax-and-Recover, licensed under the GNU General
+# Public License. Refer to the included COPYING for full text of license.
+
+if [ -n "$SSH_AGENT_PID" ]; then
```

Comment:
Why? If the user aborts the recovery and tries again it won't work.

Better leave it running till the user reboots the system. I consider the entire lifetime of the rescue system as a single user session for root.

Reply:
Hmm, we are in the `rear recover` workflow here, so `ssh` invocations from the login shell would not be aware of `ssh-agent` anyway. We simply cannot export the environment variables to the enclosing shell.

Re-trying an aborted recovery would set up another ssh-agent, wouldn't it?

Reply:
Thanks a lot for your thorough review! I'll wait for additional feedback before I come up with commits addressing outstanding issues.

Reply:
Please replace this script file with an `AddExitTask "ssh-agent -k >/dev/null"` or something like this in the `005_ssh_agent_start.sh` script. That way the agent will be killed also when ReaR aborts with an error.

Alternatively, run the agent as a system-level service that starts at boot time and writes the variables to one of the files read during login.

The benefit of that would be that 

1. you don't need to care about agent start/stop within ReaR
1. the user benefits from the agent also when he works manually with ssh outside of ReaR
1. repeated ReaR runs don't ask for the same passphrase (which was your original goal)

Reply:
Currently looking at this.

> repeated ReaR runs don't ask for the same passphrase (which was your original goal)

Actually, it wasn't about repeated ReaR runs but repeated ssh invocations during one run of `rear recover`. Over here, the backup service is accessed first to
1. list available backups and
2. to perform the actual restore.

Anyway, it would be nice to have `ssh-agent` support during the whole login session. But it shouldn't complicate things.